src/main/java/com/esri/arcgisruntime/toolkit/skins/FloorFilterSkin.java

```diff
@@ -0,0 +1,706 @@
+          }
+        }
+      });
+    }
+  }
+
+  /**
+   * A method to handle changes to the value of the isAllSitesProperty.
+   * This is toggled by the all sites checkbox option on the sites browser. It filters the list of facilities to show
+   * all facilities attached to the floor manager, as opposed to facilities attached to a selected site.
+   *
+   * @since 100.14.0
+   */
+  private void handleIsAllSitesPropertyChanged() {
+    if (isAllSitesProperty.get()) {
+      // if all sites is true, filter the facilities list only by the text field input
+      filteredFacilities.setPredicate(facility ->
+        facility.getName().toLowerCase().contains(facilitiesFilterTextField.getText().toLowerCase()));
+      // switch to displaying the facilities view for the user
+      sitesTitledPane.setExpanded(false);
+      facilitiesTitledPane.setExpanded(true);
+    } else {
+      // if all sites is false, filter the list by the text field input, and the selected site if there is one
+      if (selectedSiteProperty.get() != null) {
+        filteredFacilities.setPredicate(facility -> facility.getSite() == selectedSiteProperty.get() &&
+          facility.getName().toLowerCase().contains(facilitiesFilterTextField.getText().toLowerCase()));
+      } else {
+        filteredFacilities.setPredicate(facility ->
+          facility.getName().toLowerCase().contains(facilitiesFilterTextField.getText().toLowerCase()));
+      }
+    }
+  }
+
+  /**
+   * Method to ensure the view is not collapsed when a titled pane is expanded. A full width UI provides a better
+   * experience for browsing the titled panes.
+   *
+   * @since 100.14.0
+   */
+  private void handleTitledPaneExpanded(TitledPane titledPane) {
+    if (titledPane.isExpanded() && isCollapsedProperty.get()) {
+      isCollapsedProperty.set(false);
+    }
+  }
+
+  @Override
+  protected double computeMinWidth(double height, double topInset, double rightInset, double bottomInset, double
+    leftInset) {
+    return computePrefWidth(height, topInset, rightInset, bottomInset, leftInset);
+  }
+
+  @Override
+  protected double computeMinHeight(double width, double topInset, double rightInset, double bottomInset, double
+    leftInset) {
+    return computePrefHeight(width, topInset, rightInset, bottomInset, leftInset);
+  }
+
+  @Override
+  protected double computeMaxWidth(double height, double topInset, double rightInset, double bottomInset, double
+    leftInset) {
+    return computePrefWidth(height, topInset, rightInset, bottomInset, leftInset);
+  }
+
+  @Override
+  protected double computeMaxHeight(double width, double topInset, double rightInset, double bottomInset, double
+    leftInset) {
+    return computePrefHeight(width, topInset, rightInset, bottomInset, leftInset);
+  }
+
+  @Override
+  protected double computePrefWidth(double height, double topInset, double rightInset, double bottomInset, double
+    leftInset) {
+    if (!isCollapsedProperty.get()) {
```

Comment:
I think the way pref width is being handled prevents users from setting different pref widths so they possibly can't change the size of the control in an easy way which may or may not be what is wanted. I think they can set the min/max sizes but it isn't clear if that overrides the pref size and is definitely more awkward than just setting the one preferred size.

If you set the following in the control along with the overloads you already have then the user can set pref size to give the control different size.
```
setMaxHeight(USE_PREF_SIZE);
setMaxWidth(USE_PREF_SIZE);
setMinHeight(USE_PREF_SIZE);
setMinWidth(USE_PREF_SIZE);
```
I only remember doing this on custom rendered controls so could be there is a different/better way for controls that are just made out of standard controls.